include/dronecore.h

```diff
@@ -10,12 +10,41 @@ namespace dronecore {
 
 class DroneCoreImpl;
 
+/**
+ * DroneCore is an API and library for the PX4 flight stack using Mavlink.
+ *
+ * It is written in C++ and aiming to be:
+ *
+ * - Easy to use with a simple API.
+ * - Fast and lightweight.
+ * - Cross-platform (Linux, Mac, Windows, iOS, Android).
+ * - Extensible (using compile-time plugins).
+ *
+ */
 class DroneCore
 {
 public:
+
+    /**
+     * Constructor for the main DroneCore class
+     *
+     * In order to use DroneCore, an instance of this class needs to be created first.
+     */
     DroneCore();
+
+    /**
+     * Destructor of the main DroneCore class
+     *
+     * In order to disconnect everything and release all resources, a DroneCore instance should be
+     * destroyed.
+     */
     ~DroneCore();
 
+    /**
+     * Result returned when adding a connection.
+     *
+     * DroneCore does not throw exceptions, instead a result will be returned.
+     */
     enum class ConnectionResult {
         SUCCESS = 0,
         TIMEOUT,
```

Comment:
String for each would be good. This might not be the right place, but would be good to know for Timeout, how big the default is/where it is set. For Connections exhausted, same - what sorts of limits there are.

Reply:
You're right but it would require quite a bit of research in the code. I'll have to pass for now.